### ISSUE TITLE
Fix list-unicodeset.jsp display of RTL characters

### ISSUE THREAD

**Issue by nedley** [unicode-org/unicodetools#447]
There should be some intelligent formatting of the printed set to properly handle RTL characters.

**Comment by macchiati** [comment]
I think the list of characters is ok, but I'm guessing the UnicodeSet can
display in an odd order.

**Comment by macchiati** [comment]
Yes, that's what I was thinking of, because no processing is applied to
that.

One issue is that people copy the set, so we want to make sure that the
presentation doesn't change the contents when copied. I think that
UnicodeSet ignores RLM and LRM, but we'd have to check that.

**Comment by eggrobin** [comment]
> Sample utilities URL and what displays poorly on that page?

> Robin knows what I am talking about […]

Here is a fairly nasty one: https://util.unicode.org/UnicodeJsps/list-unicodeset.jsp?a=%5Cu0640+0-9%5C%5B&g=&i=

**Comment by eggrobin** [comment]
> For that kind of thing we could just print something like "UnicodeSet=" before the brackets, to get some strong LTR characters at the start.

That won’t help: https://util.unicode.org/UnicodeJsps/bidic.jsp?s=UnicodeSet%3D%5B%D9%80+%5C%5B+0-9%5D&b=0&u=140&d=2

What we want is to somehow have the UnicodeSet displayed according to the [Basic Ordering](https://www.unicode.org/reports/tr55/#Basic-Ordering), that is, ordering the UnicodeSet tokens left-to-right.

The naïve way to do that would be to lex the UnicodeSet syntax, and wrap every token in `<span dir="ltr">`, as described in the implementation note in that section of DUTS55.

Ned suggested another approach, which is to wrap the whole UnicodeSet in `<bdo dir="ltr">`. This should work, because the UnicodeSet tokens that can contain strongly RTL characters are always single code points.

Mark suggested yet another approach, which would have the advantage of making the UnicodeSet look right no matter where you put it, which is to apply the [Conversion to Plain Text](https://www.unicode.org/reports/tr55/#Conversion-To-Plain-Text) which would consist in inserting LRMs into the UnicodeSet, thus turning the garbled `[ـ \[ 0-9]` into `[ـ‎ \[ 0-9]`.

> I think that UnicodeSet ignores RLM and LRM

The spec says so: see `s` in https://www.unicode.org/reports/tr35/#Unicode_Sets.
Uncharacteristically, the JSPs even implement the spec properly on that point: https://util.unicode.org/UnicodeJsps/list-unicodeset.jsp?a=%5B%D9%80%E2%80%8E+%5C%5B+0-9%5D&g=&i=.

**Comment by macchiati** [comment]
I could just put a LTM after each strong RTL character (as I emit them) and
for {...} maybe just at the start and end.
